src/modules/quick_molden_module.f90

```diff
@@ -97,11 +100,11 @@ subroutine write_coordinates(self, ierr)
            ! we need to do this because optimizers may return the geometry
            ! for the next step which may be stored in xyz
            k = self%iexport_snapshot - 1
-           write(self%iMoldenFile, '(3(4x,F10.4))') (self%xyz_snapshots(j,i,k),j=1,3)
+           write(self%iMoldenFile, '(3(4x,F13.10))') (self%xyz_snapshots(j,i,k),j=1,3)
```

Comment:
This looks dangerous. F13.10 has field width 13 and 10 significant digits. Add one for the decimal point and one for  a minus sign, and this leaves only a single digit for the integer part of the coordinate.

I would use '3(2x,F20.10))' as format.